### ISSUE TITLE
Cannot open module file 'mpi.mod' for reading

### ISSUE THREAD

**Issue by Langolyers** [IllinoisRocstar/Rocstar#8]
I'm trying to install Rocstar on CentOS 7 and I get the error that mod.mpi was not found. In ccmake the module directory is correctly pointing to where the file is in `MPI_Fortran_MODULE_DIR=/usr/bin/openmpi4/lib`, but it fails when running `make -j$(nproc)`.

IMPACT failed 5 tests, all related to parallel processing, so it seems to be something with openmpi not working properly even if it finds the compilers.

[CMakeError.log](https://github.com/IllinoisRocstar/Rocstar/files/5081358/CMakeError.log)
[CMakeOutput.log](https://github.com/IllinoisRocstar/Rocstar/files/5081359/CMakeOutput.log)

**Comment by msafdari1234** [comment]
Based on what you report, it appears you have compiled IMPACT with a different compiler than the one you are trying to use for Rocstar. Usually these happen when you have multiple versions of compilers available in the environment.

**Comment by Wangduu** [comment]
My IMPACT failed those tests too, but I successfully install Rocstar. It seems that those IMPACT tests failures do not affect normal use of Rocstar.

**Comment by Langolyers** [comment]
@msafdari1234 Thanks for the reply. I tried again today from a fresh install of the OS and can confirm both IMPACT and Rocstar are using the same versions of the compilers, with the correct locations attributed within ccmake. Came up to the exact same error.

@Wangduu Could you please tell me what versions of the compilers and MPI you used? I wonder if the issue might be that since none are specified in the installation instructions. Thanks.

**Comment by Langolyers** [comment]
I tried again but this time using MPICH instead of openmpi and got a different error. It still has something to do with Fortran, but went much further into the make command. Logs are attached.

[CMakeError.log](https://github.com/IllinoisRocstar/Rocstar/files/5205364/CMakeError.log)
[CMakeOutput.log](https://github.com/IllinoisRocstar/Rocstar/files/5205365/CMakeOutput.log)

**Comment by msafdari1234** [comment]
Please double check you have posted correct logs. Based on these logs you have not progressed into build at all.

**Comment by Langolyers** [comment]
Yes, these are the correct logs. I removed everything and rebuilt from scratch with mpich for these. Error message is `testFortranCompiler.f:(.text+0xa): undefined reference to `cheev_'`. Could this be related to lapack instead? If so I might try building it from source instead of using the repositories.

**Comment by Wangduu** [comment]
@Langolyers My gcc version is 5.4.0, mpich version is 3.2, cmake version is 3.13.

**Comment by msafdari1234** [comment]
@Langolyers This is a compiler test during early stages of cmake. For some reason your compiler is incompatible. It looks like you are trying to use devtoolset-9 series compilers. They should be fine, but make sure they are the only compilers on the path and not colliding with standard system compilers.

**Comment by Langolyers** [comment]
There are no other compilers installed, I'm using a minimal install of CentOS 7. I've started from scratch with a clean installation of the OS, this time using the compilers from Development Tools (version 4.8.5) and got up to 95% in `make -j$(nproc)`. IMPACT passed all the tests as well. Logs are attached. 

[CMakeError.log](https://github.com/IllinoisRocstar/Rocstar/files/5212872/CMakeError.log)
[CMakeOutput.log](https://github.com/IllinoisRocstar/Rocstar/files/5212873/CMakeOutput.log)

**Comment by xyzxyz-321** [comment]
Have you successfully installed Impact? I encountered the following error during "make -j$(nproc)": make: *** No targets specified and no makefile found. Stop.
Can you give me some guidance? Thank you!

**Comment by Langolyers** [comment]
@xyzxyz-321 Did `cmake` run properly? Sounds like it didn't prepare the files for `make`.

Please consider opening an issue at the IMPACT repository not to mix things up.

**Comment by xyzxyz-321** [comment]
Thank you very much for answering me, but please forgive my ignorance, I did not understand what you mean.I think my problem lies in the previous step: cmake -DCMAKE_INSTALL_PREFIX=$IMPACT_INSTALL_PATH .. I don’t quite understand how to configure the file path $IMPACT_INSTALL_PATH so that when I enter: cd $IMPACT_INSTALL_PATH in the terminal, I can link to the path directly. I think this is the cause I can't find the reason for "makefile".

**Comment by msafdari1234** [comment]
If you are using bash-like environment simply set the variable by:
`export IMPACT_INSTALL_PATH=full_path_to_impact_installation_folder`
and rerun the cmake command.

**Comment by Langolyers** [comment]
@msafdari1234 Any help with my previous comment? I'm still struggling with it. Thanks.
